libminifi/test/unit/FileUtilsTests.cpp

```diff
@@ -516,3 +516,38 @@ TEST_CASE("FileUtils::path_size", "[TestPathSize]") {
 
   REQUIRE(FileUtils::path_size(dir) == 12);
 }
+
+TEST_CASE("file_clock to system_clock conversion tests") {
+  using namespace std::chrono;
+
+  static_assert(system_clock::period::num == file_clock::period::num);
+  constexpr auto lowest_den = std::min(file_clock::period::den, system_clock::period::den);
+  using LeastPreciseDurationType = duration<std::common_type_t<system_clock::duration::rep, file_clock::duration::rep>, std::ratio<system_clock::period::num, lowest_den>>;
+
+  {
+    system_clock::time_point system_now = system_clock::now();
+    file_clock::time_point converted_system_now = FileUtils::from_sys(system_now);
+    system_clock::time_point double_converted_system_now = FileUtils::to_sys(converted_system_now);
+
+    CHECK(time_point_cast<LeastPreciseDurationType>(system_now).time_since_epoch().count() == time_point_cast<LeastPreciseDurationType>(double_converted_system_now).time_since_epoch().count());
+  }
+
+  {
+    file_clock::time_point file_now = file_clock ::now();
+    system_clock::time_point converted_file_now = FileUtils::to_sys(file_now);
+    file_clock::time_point double_converted_file_now = FileUtils::from_sys(converted_file_now);
+
+    CHECK(time_point_cast<LeastPreciseDurationType>(file_now).time_since_epoch().count() == time_point_cast<LeastPreciseDurationType>(double_converted_file_now).time_since_epoch().count());
+  }
+
+  {
+    system_clock::time_point system_now = system_clock::now();
+    file_clock::time_point file_now = file_clock ::now();
+
+    file_clock::time_point converted_system_now = FileUtils::from_sys(system_now);
+    system_clock::time_point converted_file_now = FileUtils::to_sys(file_now);
+
+    CHECK(system_now-converted_file_now < 10ms);
```

Comment:
This doesn't really test what we want to test, because `system_now - converted_file_now` is either zero or negative.  The reverse should be OK:
```c++
    CHECK(converted_file_now - system_now < 10ms);
```
or, better:
```c++
    CHECK(0ms <= converted_file_now - system_now);
    CHECK(converted_file_now - system_now < 10ms);
```

Reply:
You are right, thanks. I've refactored it a bit so its more clear which one is earlier in  https://github.com/apache/nifi-minifi-cpp/pull/1560/commits/8139796ee615269c86d693942e6fc3b87d16a3bc